lib/internal/bootstrap_node.js

```diff
@@ -302,13 +302,13 @@
 
   function setupProcessFatal() {
 
-    process._fatalException = function(er) {
+    process._fatalException = function(er, fromPromise) {
       var caught;
 
       if (process.domain && process.domain._errorHandler)
         caught = process.domain._errorHandler(er) || caught;
 
-      if (!caught)
+      if (!caught && !fromPromise)
```

Comment:
@addaleax Yes. That and to make sure we don't emit an event from GC that is able to be recovered from.